ext/session/session.c

```diff
@@ -2047,13 +2047,18 @@ PHP_FUNCTION(session_set_save_handler)
 		if (register_shutdown) {
 			/* create shutdown function */
 			php_shutdown_function_entry shutdown_function_entry;
-			ZVAL_STRING(&shutdown_function_entry.function_name, "session_register_shutdown");
-			shutdown_function_entry.arg_count = 0;
-			shutdown_function_entry.arguments = NULL;
+			zval callable;
+			zend_result result;
+
+			ZVAL_STRING(&callable, "session_register_shutdown");
+			result = zend_fcall_info_init(&callable, 0, &shutdown_function_entry.fci,
+				&shutdown_function_entry.fci_cache, NULL, NULL);
+
+			ZEND_ASSERT(result == SUCCESS);
```

Comment:
This change will break some extensions, it would be better to mention it in the internal changelog :)
In addition, if ZEND_ASSERT is empty,  this may cause an unused variable warning, right?

Reply:
How can this break an extension, as if it's calling an internal function not part of itself it should declare it as a dependency.

I wouldn't worry about the compiler warning as I would expect for this to be understood by the preprocessor that it is used initially just in an empty macro, which can be optimized out.

Reply:
> as if it's calling an internal function not part of itself it should declare it as a dependency.

I tried but I can not understand...
I mean extensions need to change the code because the initialization way has changed.
In fact, there are many ZEND_API break changes now... We often receive notifications from CI that the extension built failed with php-master. But I believe all of the changes are pretty good, it just makes a little trouble for extension developers... Maybe we need to get used to these, so, forget this.